editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -276,127 +305,67 @@ void QuickOpenResultContainer::_find_candidates_in_folder(EditorFileSystemDirect
+		return;
 	}
 
-	// First, we assign a score to each candidate.
-	int num_relevant_candidates = 0;
-	for (Candidate &c : candidates) {
-		c.score = 0;
-		int prev_token_match_pos = -1;
-
-		for (const String &token : search_tokens) {
-			const int file_pos = c.file_name.findn(token);
-			const int dir_pos = c.file_directory.findn(token);
-
-			const bool file_match = file_pos > -1;
-			const bool dir_match = dir_pos > -1;
-			if (!file_match && !dir_match) {
-				c.score = -1.0f;
-				break;
-			}
-
-			float token_score = file_match ? 0.6f : 0.1999f;
-
-			// Add bias for shorter filenames/paths: they resemble the query more.
-			const String &matched_string = file_match ? c.file_name : c.file_directory;
-			int matched_string_token_pos = file_match ? file_pos : dir_pos;
-			token_score += 0.1f * (1.0f - ((float)matched_string_token_pos / (float)matched_string.length()));
-
-			// Add bias if the match happened in the file name, not the extension.
-			if (file_match) {
-				int ext_pos = matched_string.rfind(".");
-				if (ext_pos == -1 || ext_pos > matched_string_token_pos) {
-					token_score += 0.1f;
-				}
-			}
-
-			// Add bias if token is in order.
-			{
-				int candidate_string_token_pos = file_match ? (c.file_directory.length() + file_pos) : dir_pos;
-
-				if (prev_token_match_pos != -1 && candidate_string_token_pos > prev_token_match_pos) {
-					token_score += 0.2f;
-				}
-
-				prev_token_match_pos = candidate_string_token_pos;
-			}
-
-			c.score += token_score;
+	fuzzy_search->set_query(query);
+	fuzzy_search->max_results = max_total_results;
+	bool fuzzy_matching = EDITOR_GET(FUZZY_MATCHING_SETTING);
+	int max_misses = EDITOR_GET(MAX_MISSES_SETTING);
+	fuzzy_search->allow_subsequences = fuzzy_matching;
+	fuzzy_search->max_misses = fuzzy_matching ? max_misses : 0;
+
+	Vector<Ref<FuzzySearchResult>> results = fuzzy_search->search_all(filepaths);
+	candidates.resize(results.size());
+
+	for (int i = 0; i < results.size(); i++) {
+		String filename = results[i]->target;
+		StringName actual_type = *filetypes.lookup_ptr(results[i]->target);
+		QuickOpenResultCandidate candidate;
+		candidate.file_name = filename.get_file();
+		candidate.file_directory = filename.get_base_dir();
+		candidate.result = results[i];
+
+		EditorResourcePreview::PreviewItem item = EditorResourcePreview::get_singleton()->get_resource_preview_if_available(filename);
+		if (item.preview.is_valid()) {
+			candidate.thumbnail = item.preview;
+		} else if (file_type_icons.has(actual_type)) {
+			candidate.thumbnail = *file_type_icons.lookup_ptr(actual_type);
+		} else if (has_theme_icon(actual_type, EditorStringName(EditorIcons))) {
+			candidate.thumbnail = get_editor_theme_icon(actual_type);
+			file_type_icons.insert(actual_type, candidate.thumbnail);
+		} else {
+			candidate.thumbnail = *file_type_icons.lookup_ptr("__default_icon");
 		}
 
-		if (c.score > 0.0f) {
-			num_relevant_candidates++;
-		}
+		candidates.set(i, candidate);
```

Comment:
You can use `QuickOpenResultCandidate *candidated_write = candidates.ptrw();` to allow writing directly. It's faster.